ARTstor.js

```diff
@@ -1,131 +1,484 @@
 {
-	"translatorID": "5278b20c-7c2c-4599-a785-12198ea648bf",
+	"translatorID": "7a690951-ed76-43f5-b761-707fef0345d4",
 	"label": "ARTstor",
-	"creator": "Sebastian Karcher",
-	"target": "^https?://library\\.artstor\\.org",
-	"minVersion": "3.1",
+	"creator": "John Justin, Charles Zeng",
+	"target": "\\.artstor|\\.sscommons\\.org:?\\w*\\/(open)*library",
+	"minVersion": "3.0",
 	"maxVersion": "",
 	"priority": 100,
 	"inRepository": true,
 	"translatorType": 4,
 	"browserSupport": "gcs",
-	"lastUpdated": "2014-11-13 15:09:17"
-}
-
-/*
-	***** BEGIN LICENSE BLOCK *****
-	
-	ARTstor Translator, Copyright © 2012 Sebastian Karcher
-	This file is part of Zotero.
-	
-	Zotero is free software: you can redistribute it and/or modify
-	it under the terms of the GNU Affero General Public License as published by
-	the Free Software Foundation, either version 3 of the License, or
-	(at your option) any later version.
-	
-	Zotero is distributed in the hope that it will be useful,
-	but WITHOUT ANY WARRANTY; without even the implied warranty of
-	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-	GNU Affero General Public License for more details.
-	
-	You should have received a copy of the GNU Affero General Public License
-	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
-	
-	***** END LICENSE BLOCK *****
-*/
-
-function getData(doc, checkOnly) {
-	var data = {},
-		widgets = doc.getElementsByClassName('MetaDataWidget'),
-		topWidget = null, topZIndex;
-	
-	// Dtermine the top metadata window
-	for (var i=0; i<widgets.length; i++) {
-		
-		var zindex = doc.defaultView.getComputedStyle(widgets[i], null)
-			.getPropertyValue('z-index');
-		if (!zindex && zindex !== 0) continue;
-		
-		if (!topWidget || topZIndex < zindex) {
-			topWidget = widgets[i];
-			topZIndex = zindex;
-		}
-	}
-	
-	if (!topWidget) return false;
-	
-	var fields = topWidget.getElementsByClassName('field'),
-		found = false;
-	for (var i=0; i<fields.length; i++) {
-		var field = ZU.trimInternal(fields[i].textContent);
-		
-		var value = fields[i].nextElementSibling;
-		if (!value || !value.classList.contains('data')) continue;
-		value = ZU.unescapeHTML(ZU.cleanTags(value.innerHTML));
-		
-		if (!field || !value.trim()) continue;
-		
-		if (checkOnly) return true;
-		
-		found = true;
-		data[field] = value;
-	}
-	
-	return found ? data : false;
+	"lastUpdated": "2016-03-31 21:07:27"
 }
 
+/**
+    detectWeb is run to determine whether item metadata can indeed be retrieved from the webpage. 
+    The return value of this function should be the detected item type (e.g. “journalArticle”, 
+    see the overview of Zotero item types), or, if multiple items are found, “multiple”. 
+**/
 function detectWeb(doc, url) {
-	//monitor changes to body's direct children. That's where the metadata popup is added
-	Zotero.monitorDOMChanges(doc.body, {childList: true});
+    var itemType = false; // default - ignore
+    if (url.match(/\/iv2\.|ExternalIV.jsp/)) {
```

Comment:
please use `.search(/regex/) != -1` to simply test for a regular expression. More efficient than match. Also escape the period before in `.jsp`.